### PR TITLE
timespan in pizzagna implemented

### PR DESCRIPTION
**Pull request by lukasholcik**

### CODE REVIEW

File: packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.spec.ts [hunk 1]
```diff
@@ -33,4 +34,25 @@ describe("TimeseriesMetadataConfigurationComponent", () => {
...
+        let emittedValue: any = undefined;
+        timeSpanControl.setValue(null);
+        timeSpanControl.valueChanges.subscribe((value: any)=> {
+            emittedValue = value;
+        });
+
+        const timeSpans: ITimeSpanOption[] = [
+            {id: "aaa", name: "AAA"},
+            {id: "bbb", name: "BBB"}
+            ];
```
Comment (Blakesalot) [inline packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.spec.ts:51]: indentation

File: packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.ts [hunk 2]
```diff
@@ -45,16 +50,15 @@ export class TimeseriesMetadataConfigurationComponent implements IHasChangeDetec
...
                 private formBuilder: FormBuilder,
                 @Inject(PIZZAGNA_EVENT_BUS) private eventBus: EventBus<IEvent>) {
-    }
-
-    public ngOnInit(): void {
         this.form = this.formBuilder.group({
-            leftAxisLabel: [this.leftAxisLabel || ""],
-            startingTimespan: [this.startingTimespan || null, Validators.required],
-            legendPlacement: [this.legendPlacement || LegendPlacement.None, [Validators.required]],
-            preset: [this.preset || TimeseriesChartPreset.Line, [Validators.required]],
```
Comment (Blakesalot) [inline packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.ts:55]: What's the reasoning for removing these?

Comment (Blakesalot) [reply packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.ts:55]: I guess ngOnChanges takes care of the same thing?

File: packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.ts [hunk 3]
```diff
@@ -83,6 +89,12 @@ export class TimeseriesMetadataConfigurationComponent implements IHasChangeDetec
         if (changes.preset && !changes.preset.isFirstChange()) {
             this.form.get("preset")?.setValue(this.preset, {emitEvent: false});
         }
+
+        if (changes.timeSpans) {
```
Comment (Blakesalot) [inline packages/dashboards/src/lib/configurator/components/widgets/configurator-items/timeseries-metadata-configuration/timeseries-metadata-configuration.component.ts:93]: why no check for isFirstChange here?

**Review by Blakesalot — DISMISSED** [review]

**Review by Blakesalot — APPROVED** [review]

**Review by elenates — DISMISSED** [review]

**Pull request merged**